### DISCUSSION TITLE
How to remove "require.extensions is not supported by webpack. Use a loader instead" warning in Next.js project?

### DISCUSSION THREAD

**Discussion by melodyxpot** [handlebars-lang/handlebars.js#2047] [Q&A]
I got the warning when I build my next js project after I installed handlebars and started to use it. How can I remove this warning?
I appreciate your help.
This is my example screenshot
![image](https://github.com/user-attachments/assets/1d53e4d7-d22d-4a1f-9076-8df8de299c62)

Thanks in advance.

**Comment by jaylinski** [comment]
You can't remove this warning, since handlebars registers the `.hbs` and `.handlebars` file-extension:

* https://nodejs.org/api/modules.html#requireextensions
* https://github.com/handlebars-lang/handlebars.js/blob/f422bfdf3ed321c32c6eecfaa778de46275fb76d/lib/index.js#L24-L25

It's a feature. But it's deprecated, that's why webpack doesn't support it anymore.

We will probably remove it in handlebars v5.

You can try https://webpack.js.org/configuration/other-options/#ignorewarnings to remove the warning.

**Reply by mbleigh** [reply]
Is there a potential date for Handlebars v5? This is nearly a year old and I'm struggling with this problem shipping a library that depends on Handlebars.
